Automattic/sensei
# Convert the settings screen to a tabbed interface

@dwainm @danjjohnson @RiaanKnoetze -

This PR addresses the settings screen UI and UX, which currently used JavaScript and loaded all settings in a single screen.
- Adjusts the settings screen to use a tabs interface.
- Refines the loading logic to ensure only a single tab's settings are loaded for the relevant screen.
- Refines `get_settings()` to ensure the database is queried only once per load.
- Removes the unnecessary "WooThemes" branding in the settings screen header area.
- Refines `validate_settings()` to ensure all checkboxes are correctly preserved, even when not on the settings screen currently being interacted with.
## To Test
1. Load the settings screen in several browsers, to make sure the version number is correctly displayed in the header.
2. Enable checkboxes on a settings screen, and click "Save Changes". Make sure you are redirected to the tab you were interacting with.
3. Visit another settings tab, and save checkboxes and other fields there.
4. Visit the previous tab. Are your enabled checkboxes still enabled?
